src/main.rs

```diff
@@ -14,12 +14,35 @@ use std::io::Error;
 use std::sync::Arc;
 use std::thread;
 use std::thread::JoinHandle;
+
+use ::tonic::transport::Uri;
+use clap::Parser;
 use storage::content_manager::toc::TableOfContent;
 
 use crate::common::helpers::create_search_runtime;
 use crate::settings::Settings;
 
+/// Qdrant (read: quadrant ) is a vector similarity search engine.
+/// It provides a production-ready service with a convenient API to store, search, and manage points - vectors with an additional payload.
+///
+/// This CLI starts a Qdrant peer/server.
+#[derive(Parser, Debug)]
+#[clap(version, about)]
+struct Args {
+    /// Uri of the peer to bootstrap from in case of multi-peer deployment.
+    /// If not specified - this peer will be considered as a first in a new deployment.
+    #[clap(long, value_name = "URI")]
+    bootstrap: Option<Uri>,
+    /// Uri of this peer.
+    /// Other peers should be able to reach it by this uri.
+    /// Default is left for single peer deployments only.
+    #[clap(long, value_name = "URI", default_value_t=Uri::from_static("127.0.0.1:8080"))]
+    uri: Uri,
```

Comment:
It might be tricky to provide this during the deployment. Especially it might be confusing with the default value on localhost.

Could we transfer local port from the config instead, and use https://docs.rs/tonic/latest/tonic/struct.Request.html#method.remote_addr on the receiving side to detect newly joined service IP automatically?

Reply:
Ok, do you mind if we move this to a separate issue?

Reply:
ok

Reply:
https://github.com/qdrant/qdrant/issues/536